presto-spi/src/main/java/com/facebook/presto/spi/type/SqlIpAddress.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import java.util.Arrays;
+
+import static com.facebook.presto.spi.type.IpAddressType.IP_ADDRESS_SIZE;
+import static java.lang.System.arraycopy;
+import static java.util.Objects.requireNonNull;
+
+public class SqlIpAddress
+{
+    private byte[] address;
+
+    public SqlIpAddress(byte[] bytes)
+    {
+        requireNonNull(bytes, "bytes is null");
+        if (bytes.length != IP_ADDRESS_SIZE) {
```

Comment:
I'm not sure we need this constant. Sometimes you use it, other times you use `16` (a few lines below). If we're going to have it, please make sure it's used consistently everywhere.